src/proto/h1/conn.rs

```diff
@@ -1106,6 +1155,12 @@ impl State {
         self.reading = Reading::Init;
         self.writing = Writing::Init;
 
+        #[cfg(feature = "server")]
+        if self.h1_idle_timeout.is_some() {
+            // Next read will start and poll the idle timeout.
+            self.notify_read = true;
+        }
+
```

Comment:
Based on the code immediately after, this function could unconditionally set `self.notify_read = true` with a slight performance penalty if the user disables the idle timeout (the only case in which that extra polling isn't necessary).

However, the benefit would be increased maintainability, as the polling would be easier to predict (the non-polling case, which probably lacks test coverage, would be eliminated). Let me know if you want me to make this change.